I'm seeing another problem in addition to the make_strand

```
plugins/state_history_plugin/state_history_plugin.cpp:352:34:   required from ‘void eosio::state_history_plugin_impl::ws_session<SocketType>::start_read() [with SocketType = boost::asio::basic_stream_socket<boost::asio::local::stream_protocol>]’
plugins/state_history_plugin/state_history_plugin.cpp:327:42:   required from ‘void eosio::state_history_plugin_impl::ws_session<SocketType>::start() [with SocketType = boost::asio::basic_stream_socket<boost::asio::local::stream_protocol>]’
plugins/state_history_plugin/state_history_plugin.cpp:440:18:   required from ‘void eosio::state_history_plugin_impl::session_manager_t::add(eosio::state_history_plugin_impl*, std::shared_ptr<_Tp>) [with SocketType = boost::asio::basic_stream_socket<boost::asio::local::stream_protocol>]’
plugins/state_history_plugin/state_history_plugin.cpp:534:25:   required from ‘void eosio::state_history_plugin_impl::do_accept(Acceptor&) [with Acceptor = boost::asio::basic_socket_acceptor<boost::asio::local::stream_protocol>]’
plugins/state_history_plugin/state_history_plugin.cpp:519:16:   required from here
boost167/include/boost/beast/websocket/teardown.hpp:103:33: error: static assertion failed: Unknown Socket type in async_teardown.
  103 |     static_assert(sizeof(Socket)==-1,

```

--- a/plugins/state_history_plugin/state_history_plugin.cpp
+++ b/plugins/state_history_plugin/state_history_plugin.cpp
@@ -12,7 +12,6 @@
 #include <boost/asio/local/stream_protocol.hpp>
 #include <boost/asio/strand.hpp>
 #include <boost/beast/core.hpp>
-#include <boost/beast/websocket/ssl.hpp>
 #include <boost/beast/websocket.hpp>
 #include <boost/signals2/connection.hpp>
 
@@ -23,6 +22,21 @@ using tcp    = boost::asio::ip::tcp;
 using unixs  = boost::asio::local::stream_protocol;
 namespace ws = boost::beast::websocket;
 
+/* Prior to boost 1.70, if socket type is not boost::asio::ip::tcp::socket nor boost::asio::ssl::stream beast requires
+   an overload of async_teardown. This has been improved in 1.70+ to support any basic_stream_socket<> out of the box
+   which includes unix sockets. */
+#if BOOST_VERSION < 107000
+namespace boost::beast::websocket {
+template<typename TeardownHandler>
+void async_teardown(role_type, unixs::socket& sock, TeardownHandler&& handler) {
+   boost::system::error_code ec;
+   sock.close(ec);
+   boost::asio::post(boost::asio::get_associated_executor(handler, sock.get_executor()), [h=std::move(handler),ec]() mutable {
+      h(ec);
+   });
+}
+}
+#endif
 
 namespace eosio {
 using namespace chain;

